cognite/powerops/resync/config/_main.py

```diff
@@ -152,7 +152,31 @@ def load_yamls(cls, config_dir_path: Path, instantiate: bool = False) -> dict[st
 
 class CogShopConfig(Config):
     time_series_mappings: Optional[list[TimeSeriesMapping]] = None
-    watercourses_shop: list[ShopFileConfig]
+    watercourses_shop: list[ShopFileConfig]  # validation needs to happen here. If any of the files
+    _dependent_shop_files: set = {
+        "extra_data",
+        "water_value_cut_file_reservoir_mapping",
+        "water_value_cut_file",
+        "module_series",
+    }
+
+    @field_validator("watercourses_shop")
+    @classmethod
+    def validate_shop_related_files(cls, value):
+        """
+        If user has added any of the files above to the configuration, then the "cog_shop_files_config" file
+        must exist among the files to be uploaded
+        """
+        files_to_check = cls._dependent_shop_files.default.intersection({file.cogshop_file_type for file in value})
+        if not files_to_check:
+            return value
+        file_names = [file.file_path.stem for file in value]
+        seen = "cog_shop_files_config" in file_names
+        if len(files_to_check) > 1 and not seen:  # one of the files is the cog_shop_files_config itself
+            raise ValueError("Ensure that cog_shop_files_config with files order is added to watercourse directory")
+        elif len(files_to_check) == 1 and seen:
+            raise ValueError("Ensure that extra shop files to accompany cog_shop_files_config is added to watercourse")
```

Comment:
I do not understand this case. Can you elaborate in the error message?

Reply:
Since "cog_shop_files_config" also has the file_type `extra_data`, the files_to_check needs will contain the cog_shop_files_config if it is present. But if that file is present, then there needs to be other shop files as well. I realise this is not very clear. I can come up with another suggestion where the cog_shop_files_config is removed from the list before the check or something like that